config/config.go

```diff
@@ -107,13 +107,15 @@ func (c *config) Load() error {
 			c.log.Errorf("failed to merge config source: %v", err)
 			return err
 		}
-		w, err := src.Watch()
-		if err != nil {
-			c.log.Errorf("failed to watch config source: %v", err)
-			return err
+		if s, ok := src.(Source); ok {
```

Comment:
Perhaps we can take a discussion of the specific interface, for a config source, `Load()` method is necessary to implement however `Watch()` is not, so the `Watch()` method can be independent from `Source` interface, such as an interface named `Watchable`, and keep the `NewSource` function to maintain the existing usage unchanged. 

```go
type Source interface {
    Load() ([]*KeyValue, error)
}

type Watchable interface {
    Watch() (Watcher, error)
}
```

we can assert here whether config source implements the `Watchable` interface, we'd better not add a new entry function like `NewLoader`, what do you think about it?

Reply:
Yes, you are right. This is better.